### ISSUE TITLE
Current master branch not usable with Python 2

### ISSUE THREAD

**Issue by TobiasRzepka** [davidhalter/jedi#1037]
Hello,
with the current branch of jedi/parso I get always a stacktrace on Python 2!!! I believe it's since the merge of the virtualenv and master branch.
Here is my code (already known from a previous post):
```
s = jedi.Script("max_attr_value = int(2) # Cast to int for spe", 1, 45, "cmd")
s.completions()
```
With the resulting stacktrace:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python27\lib\site-packages\jedi\api\__init__.py", line 155, in completions
    self._evaluator, self._get_module(), self._code_lines,
  File "C:\Python27\lib\site-packages\jedi\api\__init__.py", line 137, in _get_module
    name = dotted_path_in_sys_path(self._evaluator.get_sys_path(), self.path)
  File "C:\Python27\lib\site-packages\jedi\evaluate\__init__.py", line 123, in get_sys_path
    return self.project._get_sys_path(self, environment=self.environment)
  File "C:\Python27\lib\site-packages\jedi\evaluate\cache.py", line 40, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Python27\lib\site-packages\jedi\api\project.py", line 96, in _get_sys_path
    sys_path = list(self._get_base_sys_path(environment))
  File "C:\Python27\lib\site-packages\jedi\api\project.py", line 83, in _get_base_sys_path
    sys_path = environment.get_sys_path()
  File "C:\Python27\lib\site-packages\jedi\cache.py", line 118, in wrapper
    result = method(self, *args, **kwargs)
  File "C:\Python27\lib\site-packages\jedi\api\environment.py", line 73, in get_sys_path
    return self._get_subprocess().get_sys_path()
  File "C:\Python27\lib\site-packages\jedi\evaluate\compiled\subprocess\__init__.py", line 162, in get_sys_path
    return self._send(None, functions.get_sys_path, (), {})
  File "C:\Python27\lib\site-packages\jedi\evaluate\compiled\subprocess\__init__.py", line 201, in _send
    is_exception, traceback, result = pickle_load(self._process.stdout)
  File "C:\Python27\lib\site-packages\jedi\_compatibility.py", line 384, in pickle_load
    return pickle.load(file)
  File "C:\Python27\lib\pickle.py", line 1384, in load
    return Unpickler(file).load()
  File "C:\Python27\lib\pickle.py", line 864, in load
    dispatch[key](self)
KeyError: '\n'
```
Tobias

**Comment by davidhalter** [comment]
I haven't tested it on Windows, yet. glad I have a tester now :) I'll look into it and get back to you.

I'm pretty sure there's a few more problems there. Does Python 3 work for you or are you not using it?

**Comment by TobiasRzepka** [comment]
Hello Dave,Because I'm on work, I'll ‎answer directly: on Python 3 there's no such Problem. Tobias                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Von: Dave HalterGesendet: Freitag, 2. Februar 2018 09:36An: davidhalter/jediAntwort an: davidhalter/jediCc: TobiasRzepka; AuthorBetreff: Re: [davidhalter/jedi] Current master branch not usable with Python 2 (#1037)I haven't tested it on Windows, yet. glad I have a tester now :) I'll look into it and get back to you.
I'm pretty sure there's a few more problems there. Does Python 3 work for you or are you not using it?

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/davidhalter/jedi","title":"davidhalter/jedi","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/davidhalter/jedi"}},"updates":{"snippets":[{"icon":"PERSON","message":"@davidhalter in #1037: I haven't tested it on Windows, yet. glad I have a tester now :) I'll look into it and get back to you.\r\n\r\nI'm pretty sure there's a few more problems there. Does Python 3 work for you or are you not using it?"}],"action":{"name":"View Issue","url":"https://github.com/davidhalter/jedi/issues/1037#issuecomment-362520771"}}}</scri
 pt><br><!--end of _originalContent --></div>

**Comment by davidhalter** [comment]
Is there some output when you use `python2 -m jedi.evaluate.compiled.subprocess XXX`?

Replace XXX with the output of 
```
python2 -c 'import parso, os; print(os.path.dirname(os.path.dirname(parso.__file__)))'
```.

It looks like something is printed to stdout.

**Comment by TobiasRzepka** [comment]
I tried your suggestion but it shows nothing at all. I got a prompt waiting for pickled input, because if I enter something, the stacktrace looks the same (guess I'm not a fluent speaker of pickle :).
Do you have any other suggestions I can get deeper into it?

**Comment by davidhalter** [comment]
Fixed by #1037. Will merge later. Thanks a lot!
